### ISSUE TITLE
SignAssembly seems to always delay-sign regardless of the presence or value of DelaySign

### ISSUE THREAD

**Issue by peteroupc** [microsoft/msbuild#2503]
In MSBuild 15.3, specifying `<SignAssembly>True</SignAssembly>` in a .csproj project&mdash;

- but without DelaySign results in a delay-signed, not strong-named assembly.
- and with `<DelaySign>False</DelaySign>` has the same result.
- and with `<DelaySign>True</DelaySign>` has the same result.

Thus there seems to be no way to strong-name an assembly without running an external tool to do so.

(Note that in all these cases, `AssemblyOriginatorKeyFile` is present and set to an existing strong name key file.)

**Comment by iamthegD** [comment]
I'm having the same problem. 

We are extending RegisterAssembly in the msbuild sln via a new project and  and the result is never strong named. (added on top of 15.3)

Will try to manually sn.

**Comment by craigfowler** [comment]
I have also reproduced this on MSBuild for Mono/Linux:

## Version info
```
$ msbuild /ver
Microsoft (R) Build Engine version 15.4.0.0 ( Wed Nov 29 14:56:11 UTC 2017) for Mono
Copyright (C) Microsoft Corporation. All rights reserved.

15.4.0.0
```

## My project
In my `.csproj` file I have the following snippet:

```xml
<PropertyGroup>
  <SignAssembly>true</SignAssembly>
  <AssemblyOriginatorKeyFile>..\MyKeyfile.snk</AssemblyOriginatorKeyFile>
  <DelaySign>false</DelaySign>
</PropertyGroup>
```

I have tried this with `DelaySign` set to false or with the element removed altogether.  If I use msbuild to build the project though and then take a look at one of my compiled assemblies, I see this:

```
$ sn -vf MyAssembly/bin/Debug/MyAssembly.dll 
Mono StrongName - version 5.10.0.160
StrongName utility for signing assemblies
Copyright 2002, 2003 Motus Technologies. Copyright 2004-2008 Novell. BSD licensed.

Assembly MyAssembly/bin/Debug/MyAssembly.dll is delay-signed but not strongnamed
```

According to my understanding, msbuild should fully-sign the assembly and not delay-sign it when the delay-sign property is set to false.

## Workaround: Manually sign the delay-signed assembly via sn.exe
When msbuild fails to sign & strong-name my assembly, following a build up with this does get the job done though:
```
$ sn -R MyAssembly/bin/Debug/MyAssembly.dll MyKeyfile.snk && sn -vf MyAssembly/bin/Debug/MyAssembly.dll
Mono StrongName - version 5.10.0.160
StrongName utility for signing assemblies
Copyright 2002, 2003 Motus Technologies. Copyright 2004-2008 Novell. BSD licensed.

Assembly MyAssembly/bin/Debug/MyAssembly.dll signed.
Mono StrongName - version 5.10.0.160
StrongName utility for signing assemblies
Copyright 2002, 2003 Motus Technologies. Copyright 2004-2008 Novell. BSD licensed.

Assembly MyAssembly/bin/Debug/MyAssembly.dll is strongnamed.
```

**Comment by rainersigwald** [comment]
@craigfowler You're seeing a mono-specific bug; I'll respond in your mono/mono#8495.

Other folks: are you also using Mono? If not, my attempts to reproduce this failed. Can you provide more details or a repro project?

**Comment by peteroupc** [comment]
@rainersigwald : I was indeed using Mono when I saw the issue I reported.

**Comment by craigfowler** [comment]
At least from my end then, I'd be happy to call this resolved ("Can't repro") in this repo at least.  It's a mono-specific problem which apparently is already fixed in upcoming Mono 5.14.

@peteroupc - I've opened [this same issue against the mono repo] if you want to track it there, it being a mono-specific problem.  In short though: I'm working around it using `sn -R` manually in my builds for the time being.

[this same issue against the mono repo]: https://github.com/mono/mono/issues/8495

**Comment by Mizux** [comment]
my 2 cents:
using: `<PublicSign>False</PublicSign>` seems to fix the always delay signed issue...
